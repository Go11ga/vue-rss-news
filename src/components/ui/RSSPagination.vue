<template>
    <div v-if="totalCount !== 0" class="rss-pagination">

        <RSSPaginationItem label="«" :value="1" size="big" />

        <RSSPaginationItem label="‹" :value="currentPage > 1 ? currentPage - 1 : 1" size="big" />

        <RSSPaginationItem
            v-for="(page, index) in pages"
            :key="index"
            :label="page.label"
            :value="page.value"
            :active="page.value == currentPage"
        />

        <RSSPaginationItem label="›" :value="currentPage < totalCount ? currentPage + 1 : totalCount" size="big" />

        <RSSPaginationItem label="»" :value="totalCount" size="big" />

    </div>
</template>

<script>
import RSSPaginationItem from '@/components/ui/RSSPaginationItem'
import getPaginationOptions from '@/utils/pagiantionOptions'

export default {
    name: 'RSSPagination',

    props: {
        currentPage: {
            type: Number,
            required: true
        },
        totalCount: {
            type: Number,
            required: true
        }
    },

    components: {
        RSSPaginationItem
    },

    data: () => ({
        paginationLength: 8
    }),

    computed: {
        pages() {
            return getPaginationOptions({
                listLength: this.paginationLength,
                currentPage: this.currentPage,
                totalPages: this.totalCount
            })
        }
    }
}
</script>