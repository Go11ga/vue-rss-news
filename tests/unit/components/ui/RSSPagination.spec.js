import { shallowMount } from '@vue/test-utils'
import RSSPagination from '@/components/ui/RSSPagination'

describe('RSSPagination.vue', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(RSSPagination, {
            propsData: {
                currentPage: 1,
                totalCount: 8
            }
        })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('проверка имени компонента', () => {
        expect(wrapper.vm.$options.name).toMatch('RSSPagination')
    })

    it('snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it('computed pages', () => {
        let localThis = {
            paginationLength: 3,
            currentPage: 1,
            totalCount: 3
        }

        expect(RSSPagination.computed.pages.call(localThis)).toStrictEqual([
            { label: '1', value: 1 },
            { label: '2', value: 2 },
            { label: '3', value: 3 }
        ])
    })
})