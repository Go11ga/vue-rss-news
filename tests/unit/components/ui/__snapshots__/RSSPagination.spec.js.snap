// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSSPagination.vue snapshot 1`] = `
<div
  class="rss-pagination"
>
  <rsspaginationitem-stub
    label="«"
    size="big"
    value="1"
  />
   
  <rsspaginationitem-stub
    label="‹"
    size="big"
    value="1"
  />
   
  <rsspaginationitem-stub
    active="true"
    label="1"
    value="1"
  />
  <rsspaginationitem-stub
    label="2"
    value="2"
  />
  <rsspaginationitem-stub
    label="3"
    value="3"
  />
  <rsspaginationitem-stub
    label="4"
    value="4"
  />
  <rsspaginationitem-stub
    label="5"
    value="5"
  />
  <rsspaginationitem-stub
    label="6"
    value="6"
  />
  <rsspaginationitem-stub
    label="7"
    value="7"
  />
  <rsspaginationitem-stub
    label="8"
    value="8"
  />
   
  <rsspaginationitem-stub
    label="›"
    size="big"
    value="2"
  />
   
  <rsspaginationitem-stub
    label="»"
    size="big"
    value="8"
  />
</div>
`;
